### DISCUSSION TITLE
Fish support

### DISCUSSION THREAD

**Discussion by tristanRW** [wavetermdev/waveterm#779] [General]
Is fish support planned for wave ? What would be issues to tackle before it is possible ?

**Comment by esimkowitz** [comment]
Hey, Fish is now officially supported in Wave. Let me know how it works for you!

**Reply by texastoland** [reply]
But it's "supported" because blocks are now unsupported right? Will they make a return? My assumption is the complications documented by the Warp team weren't worth the implementation cost or burden to support additional shells.

Unrelated but Wave really needs a system prompt. I asked `get mac os version` and it suggested `About This Mac`. I prefixed it with `term cmd` to get `sw_vers`. I imagine it's on a list but it'd be nice to edit and rerun as well as clear the chat.

**Reply by esimkowitz** [reply]
Correct, yeah the command blocks are probably not making a comeback in the near future. The feedback we got from users was that the state management stuff we were doing, while cool, was breaking compatibility with too many tools they expected to "just work". Chasing state management parity on every shell someone would want to use was not something we could afford to do at our stage and size. We are working to bring new features that we think will bridge the gap, such as pinning a specific command to a block (a layout element in the new app, different from the command blocks in the old app) so that you can click to rerun it and place it somewhere in your layout where you can always reach it.

As for the second suggestion, are you referring to Wave AI? If so, then yes I totally agree. We're working on integrating the OpenAI context mechanism, which will allow us to dynamically update system values with each request. We're also looking into using the functions mechanism to allow the agent to request even more specific information about your configuration

**Reply by texastoland** [reply]
> The feedback we got from users was that the state management stuff we were doing, while cool, was breaking compatibility with too many tools they expected to "just work".

FWIW +1. I love Warp but it never played well with Nushell etc. either. Wave's widgets also solve more "real-world" use cases (for me anyway). The editor is esp. 🔥


> so that you can click to rerun it and place it somewhere in your layout where you can always reach it.

My exact use case was editing a typo and rerunning it. Just rerunning sounds useful for scripts though 💯

> We're working on integrating the OpenAI context mechanism, which will allow us to dynamically update system values with each request.

Exactly this 🙌🏼

**Comment by tristanRW** [comment]
Will the command history be making a comeback?
I think warp does that well, where you can filter each block using regex. Since wave does not have blocks anymore, I could imagine it working similar to the previous global history - maybe with the addition of regex?
